docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+Here, we use the term naive to describe a tool that assumes perfect input in a known structure. 
+Later we will add input fields that accept arbitrary values. But we can start with controlled, known values first to simplify testing.
+To further simplify the transition to User populated or arbitrary data, we can also create a constants.py file.
+For now, leave it blank. We'll need it later to create static values for our tests.
+
+We update our tool by "mode" (recall our modes are: `DATA, PREVIEW, TRAIN, PREDICT`).
+Considering we need to test building the model in that order anyway, this will make reasoning about our code more straightforward.
+
+
+### 1.3.1. Data Mode
+
+For this guide, we'll be using the [imdb movie review dataset](https://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz).
+This archive is a dataset of movie reviews and their labels for classification.
+The labels are binary positive or negative reviews.
+The reviews are scrapped from the IMDb website and then appropriately labeled.
+You may download it now or during the plugin runtime.
+Here we download it and reference it locally.
+Note that we get this in `.tar.gz` format, so you need to decompress this before using the data.
+Once you decompress, note the location as a constant in the `constants.py` module we created earlier.
+Something like this will do:
+
+`DATA_SRC = '/path/to/archive/aclImdb_v1.tar/'`
+
+Next, open up `backend/ayx_plugins/text_classifier.py`. It's finally time to add some functional code!
+
+Update the `__init__` method to the below:
+
+```python
+    def __init__(self, provider: "AMPProviderV2") -> None:
+        """Construct a plugin."""
+        self.provider = provider
+
+        self.MODE = "DATA"
+
+        # The location of the raw data
+        self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]
+        # The location to store training datasets for our model
+        self.train_dir = self.provider.tool_config["datasetConfig"]["trainingSetDir"]
+        # Test datasets, like the training above.
+        self.test_dir = self.provider.tool_config["datasetConfig"]["testSetDir"]
+        # We pull validation datasets from the same location as our training here
+        self.val_dir = self.train_dir
+
+        self.provider.io.info("Plugin initialized.")
+```
+
+Where:
+
+`self.MODE = "DATA"`
+ is our tool's "run mode," as described earlier in [link to section].
+
+And:
+
+ `self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]` 
+
+ And the lines below it utilize the `tool_config` dict (link to tool_config in docs) to access data and tool configuration sent by the UI SDK.
+
+You may have already noticed, but `tool_config` will contain the values we said we'd mock before deploying to Designer.
+Once we have developed our frontend, a user can populate these fields.
+Until then, you can safely use constants for testing. (better phrasing?)
+
+Next, add the following to the `on_complete` method:
```

Comment:
TW review:

Where `self.MODE = "DATA"` is the tool's "run mode," as described earlier in [link to section], and `self.data_url = self.provider.tool_config["datasetConfig"]["datasetTargetDir"]` (and the lines below it) utilize the `tool_config` dict (link to tool_config in docs) to access data and tool configuration sent by the UI SDK.

You might have already noticed, but `tool_config` contains the values we said we'd mock before we deploy to Alteryx Designer. Once we develop our front end, a user can populate these fields. Until then, you can safely use constants for testing.

Next, add this to the `on_complete` method:

Reply:
There is also a [link to section] item that needs to be updated.